src/manifest/mod.rs

```diff
@@ -34,6 +36,157 @@ struct CargoPackage {
+pub struct CargoWasmPackProfile {
+    #[serde(default, rename = "wasm-bindgen")]
+    wasm_bindgen: CargoWasmPackProfileWasmBindgen,
+}
+
+#[derive(Default, Deserialize)]
+struct CargoWasmPackProfileWasmBindgen {
+    #[serde(default, rename = "debug-js-glue")]
+    debug_js_glue: Option<bool>,
+
+    #[serde(default, rename = "demangle-name-section")]
+    demangle_name_section: Option<bool>,
+
+    #[serde(default, rename = "dwarf-debug-info")]
+    dwarf_debug_info: Option<bool>,
+}
+
+impl CargoWasmPackProfile {
+    fn default_dev() -> Self {
+        CargoWasmPackProfile {
+            wasm_bindgen: CargoWasmPackProfileWasmBindgen {
+                debug_js_glue: Some(true),
+                demangle_name_section: Some(true),
+                dwarf_debug_info: Some(false),
+            },
+        }
+    }
+
+    fn default_release() -> Self {
+        CargoWasmPackProfile {
+            wasm_bindgen: CargoWasmPackProfileWasmBindgen {
+                debug_js_glue: Some(false),
+                demangle_name_section: Some(true),
+                dwarf_debug_info: Some(false),
+            },
+        }
+    }
+
+    fn default_profiling() -> Self {
+        CargoWasmPackProfile {
+            wasm_bindgen: CargoWasmPackProfileWasmBindgen {
+                debug_js_glue: Some(false),
+                demangle_name_section: Some(true),
+                dwarf_debug_info: Some(false),
+            },
+        }
+    }
+
+    fn deserialize_dev<'de, D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let mut profile = <Option<Self>>::deserialize(deserializer)?.unwrap_or_default();
+        profile.update_with_defaults(Self::default_dev());
+        Ok(profile)
+    }
+
+    fn deserialize_release<'de, D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let mut profile = <Option<Self>>::deserialize(deserializer)?.unwrap_or_default();
+        profile.update_with_defaults(Self::default_release());
+        Ok(profile)
+    }
+
+    fn deserialize_profiling<'de, D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let mut profile = <Option<Self>>::deserialize(deserializer)?.unwrap_or_default();
+        profile.update_with_defaults(Self::default_profiling());
+        Ok(profile)
+    }
+
+    fn update_with_defaults(&mut self, defaults: Self) {
+        macro_rules! d {
+            ( $( $path:ident ).* ) => {
+                self. $( $path ).* .get_or_insert(defaults. $( $path ).* .unwrap());
+            }
+        }
+        d!(wasm_bindgen.debug_js_glue);
+        d!(wasm_bindgen.demangle_name_section);
+        d!(wasm_bindgen.dwarf_debug_info);
+    }
+
+    /// Get this profile's configured `[wasm-bindgen.debug-js-glue]` value.
+    pub fn wasm_bindgen_debug_js_glue(&self) -> bool {
+        self.wasm_bindgen.debug_js_glue.unwrap()
```

Comment:
I leaned back towards them because making the two-struct approach work with nested structs involves a lot of boilerplate and it isn't totally clear how we would know which defaults to use in the nested struct deserialization where we no longer know which profile we are deserializing for.

Reply:
Ok! Just wanted to confirm. We can always tweak over time if necessary!